### PR TITLE
map all non-singleton bindings as prototype scoped beans

### PR DESCRIPTION
**Pull request by twicksell**
Resolution for #57

### CODE REVIEW

File: src/test/java/org/springframework/guice/ScopingTests.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		assertNotNull(someNoScopeDependency1);
+		assertNotNull(someNoScopeDependency2);
+		assertNotEquals(someNoScopeDependency1, someNoScopeDependency2);
+		
+		SomeCustomScopeDependency someCustomScopeDependency1 = context.getBean(SomeCustomScopeDependency.class);
+		SomeCustomScopeDependency someCustomScopeDependency2 = context.getBean(SomeCustomScopeDependency.class);
+		
+		assertNotNull(someCustomScopeDependency1);
+		assertNotNull(someCustomScopeDependency2);
+		assertEquals(someCustomScopeDependency1, someCustomScopeDependency2);
```
Comment (asibross) [inline src/test/java/org/springframework/guice/ScopingTests.java]: Shouldn't this be NotEqual? I thought custom scopes should not be singletons.

File: src/main/java/org/springframework/guice/annotation/ModuleRegistryConfiguration.java [hunk 2]
```diff
@@ -118,8 +120,12 @@ private void mapBindings(Map<Key<?>, Binding<?>> bindings,
 			ConstructorArgumentValues args = new ConstructorArgumentValues();
 			args.addIndexedArgumentValue(0, key.getTypeLiteral().getRawType());
 			args.addIndexedArgumentValue(1, key);
+			args.addIndexedArgumentValue(2, Scopes.isSingleton(binding));
 			bean.setConstructorArgumentValues(args);
 			bean.setTargetType(ResolvableType.forType(key.getTypeLiteral().getType()));
+			if(!Scopes.isSingleton(binding)) {
+				bean.setScope(ConfigurableBeanFactory.SCOPE_PROTOTYPE);
```
Comment (asibross) [inline src/main/java/org/springframework/guice/annotation/ModuleRegistryConfiguration.java]: For completeness shouldn't we leverage Spring Scope interface and create an adapter from the Guice custom scope?

File: src/test/java/org/springframework/guice/ScopingTests.java (hunk 1 above)
Comment (twicksell) [reply src/test/java/org/springframework/guice/ScopingTests.java]: I implemented my custom scope to be a singleton. Guess I can make it be prototype.

File: src/main/java/org/springframework/guice/annotation/ModuleRegistryConfiguration.java (hunk 2 above)
Comment (twicksell) [reply src/main/java/org/springframework/guice/annotation/ModuleRegistryConfiguration.java]: The only pre-defined scopes in both frameworks are Singleton and Prototype. I can't think of any way to better handle any of the user defined scopes in either framework.

**Pull request merged**